### ISSUE TITLE
Add --stdin flag

### ISSUE THREAD

**Issue by alypeng** [stylelint/stylelint#4570]
> Clearly describe the bug

If `a.css` is an empty file then `cat a.css | stylelint` outputs help instead of an error.

> Which rule, if any, is the bug related to?

`no-empty-source`

> What CSS is needed to reproduce the bug?

An empty file (e.g. `touch a.css`)

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-empty-source": true
  }
}
```

> Which version of stylelint are you using?

13.0.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

Same output as `stylelint a.css`

```
a.css
 1:1  ✖  Unexpected empty source   no-empty-source
```

> What actually happened (e.g. what warnings or errors did you get)?

Same output as `stylelint --help`

```

  A mighty, modern CSS linter.

  Usage: stylelint [input] [options]

  Input: Files(s), glob(s), or nothing to use stdin.

    If an input argument is wrapped in quotation marks, it will be passed to
    globby for cross-platform glob support. node_modules are always ignored.
    You can also pass no input and use stdin, instead.

  Options:

...
```

Possibly related to #1557

**Comment by jeddy3** [comment]
@alypeng Thanks for the report and for using the template.

>> What did you expect to happen?

> Same output as `stylelint a.css`

That does seem like the best behaviour. I suspect ESLint has its `--stdin` flag to support this behaviour. Without the flag ESLint, like stylelint, prints help when an empty file is piped to it.

Ideally, we can achieve this without a new flag because turning off `stdin` support by default would be a breaking change.

I'll label as a discussion to see if anyone else has an opinion on this.

**Comment by jeddy3** [comment]
I suggest we implement this as a non-breaking change. We can keep the existing default behaviour, but add a new `--stdin` flag that changes the behaviour of the CLI to not trigger help if an empty file it piped in. 

We would document the flag in the [Developer guide > Usage > CLI (options)](https://stylelint.io/user-guide/usage/cli#options), perhaps as:

```md
### `--stdin`

Force accepting of the standard input stream even if it is empty
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.
